### PR TITLE
feat: 🎸 implement status modal

### PR DESCRIPTION
**Pull request by crewneck**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -61,6 +61,7 @@ import SegmentedControl from './Molecules/SegmentedControl';
 import Select from './Molecules/Select';
 import SelectionCard from './Molecules/SelectionCard';
 import Slider from './Molecules/Slider';
+import StatusModal from './Molecules/StatusModal';
```
Comment (mjmonline) [inline src/index.ts:64]: This component reuses other molecules, so the idea is that it should then be a organism. I'm not a fan of current structure but as long as we have it let's try to stick to it 🙂

File: src/Molecules/StatusModal/StatusModal.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  .add('Loading', () => <StatusModal {...defaultProps} loading />)
+  .add('With all', () => <StatusModal {...defaultProps} status="CASE_1" options={options} />)
+  .add('Success', () => (
+    <StatusModal {...defaultProps} status="SUCCESS" options={options} />
+  ))
+  .add('Warning', () => (
+    <StatusModal {...defaultProps} status="WARNING" options={options} />
+  ))
+  .add('Error', () => (
+    <StatusModal {...defaultProps} status="ERROR" options={options} />
```
Comment (mjmonline) [inline src/Molecules/StatusModal/StatusModal.stories.tsx:44]: I think this API feels weird, so you pass in a object as `options` props and then with `status` prop you decide which of the `options` keys to use? In my opinion there is no need of letting the component handle this logic you could just pass the object you wanted in the first place like `options={options['SUCCESS']}` right?

Also this way of doing it devs have to always define the statuses for example that success needs to have the icon `<Icon.CheckMarkCircle color={(t) => t.color.positive} size={23} />` every time. And so on for the rest of the statuses

Comment (LifeofMichal) [reply src/Molecules/StatusModal/StatusModal.stories.tsx:44]: I found it weird too, but from what I've heard this is how Status Modal is implemented most of the time in UIM.

File: src/index.ts (hunk 1 above)
Comment (LifeofMichal) [reply src/index.ts:64]: That's a good point 👍🏻

File: src/Molecules/StatusModal/StatusModal.stories.tsx (hunk 2 above)
Comment (crewneck) [reply src/Molecules/StatusModal/StatusModal.stories.tsx:44]: @mjmonline I think I see what you mean. I'll try to strip out more of what we send in to the component.

**Review by LifeofMichal — APPROVED** [review]

**Review by jesperwiner — APPROVED** [review]

**Pull request merged**